assets/js/payment-method-extensions/payment-methods/stripe/payment-request/payment-request-express.js

```diff
@@ -0,0 +1,380 @@
+/**
+ * Internal dependencies
+ */
+import {
+	DEFAULT_STRIPE_EVENT_HANDLERS,
+	PAYMENT_METHOD_NAME,
+} from './constants';
+import {
+	getStripeServerData,
+	getPaymentRequest,
+	updatePaymentRequest,
+	canDoPaymentRequest,
+	getTotalPaymentItem,
+	getBillingData,
+	getPaymentMethodData,
+	getShippingData,
+	normalizeShippingAddressForCheckout,
+	normalizeShippingOptions,
+	normalizeLineItems,
+	normalizeShippingOptionSelectionsForCheckout,
+} from '../stripe-utils';
+
+/**
+ * External dependencies
+ */
+import { useRef, useState, useEffect } from '@wordpress/element';
+import {
+	Elements,
+	PaymentRequestButtonElement,
+	useStripe,
+} from '@stripe/react-stripe-js';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * @typedef {import('../stripe-utils/type-defs').Stripe} Stripe
+ * @typedef {import('../stripe-utils/type-defs').StripePaymentRequest} StripePaymentRequest
+ * @typedef {import('@woocommerce/type-defs/registered-payment-method-props').RegisteredPaymentMethodProps} RegisteredPaymentMethodProps
+ */
+
+/**
+ * @typedef {Object} WithStripe
+ *
+ * @property {Stripe} [stripe] Stripe api (might not be present)
+ */
+
+/**
+ * @typedef {RegisteredPaymentMethodProps & WithStripe} StripeRegisteredPaymentMethodProps
+ */
+
+/**
+ * PaymentRequestExpressComponent
+ *
+ * @param {StripeRegisteredPaymentMethodProps} props Incoming props
+ */
+const PaymentRequestExpressComponent = ( {
+	shippingData,
+	billing,
+	eventRegistration,
+	onSubmit,
+	activePaymentMethod,
+	setActivePaymentMethod,
+	setExpressPaymentError,
+} ) => {
+	/**
+	 * @type {[ StripePaymentRequest|null, function( StripePaymentRequest ):StripePaymentRequest|null]}
+	 */
+	// @ts-ignore
+	const [ paymentRequest, setPaymentRequest ] = useState( null );
+	const stripe = useStripe();
+	const [ canMakePayment, setCanMakePayment ] = useState( false );
+	const [ paymentRequestType, setPaymentRequestType ] = useState( '' );
+	const [ isProcessing, setIsProcessing ] = useState( false );
+	const [ isFinished, setIsFinished ] = useState( false );
+	const eventHandlers = useRef( DEFAULT_STRIPE_EVENT_HANDLERS );
+	const currentBilling = useRef( billing );
+	const currentShipping = useRef( shippingData );
+	const currentPaymentRequest = useRef( paymentRequest );
+	// for keeping track of what the active payment method was before this
+	// payment button was clicked.
+	const originalActivePaymentMethod = useRef( activePaymentMethod );
+
+	// update refs when any change
+	useEffect( () => {
+		currentBilling.current = billing;
+		currentShipping.current = shippingData;
+		currentPaymentRequest.current = paymentRequest;
+	}, [ billing, shippingData, paymentRequest ] );
+
+	//set paymentRequest
```

Comment:
```suggestion
	// set paymentRequest
```

Reply:
fixed this and other things (missing period) in 0b03b0b